ext/sqlite3/database.c

```diff
@@ -30,16 +37,22 @@ database_memsize(const void *ctx)
     return sizeof(*c);
 }
 
+static void
+database_update_references(void *ctx)
+{
+    sqlite3RubyPtr c = (sqlite3RubyPtr)ctx;
+    c->busy_handler = rb_gc_location(c->busy_handler);
+}
+
 static const rb_data_type_t database_type = {
-    "SQLite3::Backup",
-    {
-        NULL,
-        deallocate,
-        database_memsize,
+    .wrap_struct_name = "SQLite3::Backup",
```

Comment:
Another question from ignorance:

Peter's article describes the `wrap_struct_name` this way:
>This is a C string name we give for our TypedData object. This name is used for debugging and statistics purposes. Although there are no requirements for this name (it just has to be a valid C string that’s null-terminated), it is a good idea to give it an informative and unique name to make it easy for yourself and other developers to identify the object.

For my own edification, why do we use the same string for 3 different TypedData objects:

https://github.com/search?q=repo%3Asparklemotion%2Fsqlite3-ruby%20rb_data_type_t&type=code

Reply:
No idea. I personally tend to set it as the name of the class it backs, but as Peter says, it can be anything.

Ultimately this doesn't have a big impact, the only way to see it is via `ObjectSpace.dump` and `ObjectSpace.dump_all` so is only useful when debugging memory related things.

Would probably be worth reworking them all, but I don't own that gem, so not my call.